web/modules/custom/ecc_cct/src/Controller/ConsentLog.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\ecc_cct\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Returns responses for ECC Cookie Consent Tracker routes.
+ */
+final class ConsentLog extends ControllerBase {
+
+  /**
+   * Handles requests to /ecc-cct/js/log/{$choice}.
+   *
+   * @param int $choice
+   *   The dynamic ID passed to the route.
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The HTTP request object.
+   *
+   * @return \Symfony\Component\HttpFoundation\Response
+   *   A response object.
+   */
+  public function log(int $choice, Request $request): Response {
+    // If this isn't turned on, bail.
+    if (!\Drupal::config('ecc_cct.settings')->get('enable')) {
+      throw new ServiceUnavailableHttpException(NULL, t('Service unavailable'));
+    }
+    // Otherwise, let's write some data.
+    $content = [
+      'status' => 'success',
+      'message' => "Received ID: $choice",
+    ];
+
+    /** @var \Drupal\Core\Database\Connection $connection */
+    $connection = \Drupal::service('database');
```

Comment:
This should use a content entity instead of directly accessing the database table. It would make CRUD much easier.

Reply:
OK. Don't call services like this. Use create() for dependency injection.